CacheSim/CacheSim.h

```diff
@@ -28,25 +28,63 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <stdint.h>
 #include <stdio.h>
+#if defined(_MSC_VER)
 #include <Windows.h>
+#ifndef IG_CACHESIM_API
+#define IG_CACHESIM_API __declspec(dllimport)
+#endif
+#else
+#include <dlfcn.h>
+#include <unistd.h>
+#define IG_CACHESIM_API
```

Comment:
Right by default on Linux all symbols are exported, which sucks. Best practice is to make them all hidden via command line switch (`-fdefault-visibility=hidden` or something like that) and explicitly expose the ones you want `dlopen` and friends to worry about.